examples/qt/lineplot.py

```diff
@@ -0,0 +1,317 @@
+"""
+Lineplot Qt
+===========
+
+Complex example for lineplot in PyQt that displays 3 traces.
+The plot is a standard black on white with a legend on the right.
+
+"""
+import sys
+import numpy as np
+import time
+from math import pi, cos, sin, ceil, log10
+
+import fastplotlib as fpl
+
+try:
+    from PyQt6.QtWidgets import QApplication, QWidget, QVBoxLayout, QMainWindow
+    from PyQt6.QtCore import QTimer, Qt
+
+except ImportError:
+    from PyQt5.QtWidgets import QApplication, QWidget, QVBoxLayout, QMainWindow
+    from PyQt5.QtCore import QTimer, Qt
+
+def rotate(angle, axis_x, axis_y, axis_z):
+    """
+    Quaternion representing rotation around the given axis by the given angle.
+    """
+    a2 = angle/2.0
+    c = cos(a2)
+    s = sin(a2)
+    return (axis_x * s, axis_y * s, axis_z * s, c)
+
+class FastPlotMain(QMainWindow):
+
+    MAJOR_TICKS = 5
+    MINOR_TICKS = 4
+    DATAPOINTS  = 5000                   # number of data points per line
+    INTERVAL    = 16                     # display refresh in milliseconds
+    WHITE       = (1.0, 1.0, 1.0, 1.0)
+    BLACK       = (0.0, 0.0, 0.0, 1.0)
+    RED         = (1.0, 0.0, 0.0, 1.0)
+    GREEN       = (0.0, 1.0, 0.0, 1.0)
+    BLUE        = (0.0, 0.0, 1.0, 1.0)
+    DARK_GRAY   = (0.2, 0.2, 0.2, 1.0)
+    LIGHT_GRAY  = (0.9, 0.9, 0.9, 1.0)
+
+    def __init__(self):
+        super().__init__()
+
+        # ─── Window ──────────────────────────────────────────────────────────────
+        
+        self.setWindowTitle("fastplotlib Line Plot Test")
+        self.resize(800, 600)
+
+        # ─── Figure & Subplot ────────────────────────────────────────────────────
+
+        self.fig = fpl.Figure(
+            (1, 1),
+            size=(800, 600),
+            names = "Line Plot",
+        )
+
+        # Subplot
+        self.ax = self.fig[0, 0]
```

Comment:
let's just call these subplot instead of `ax` to conform with the rest of the examples, I can make these changes if you want